app/dashboard/views.py

```diff
@@ -5348,10 +5349,25 @@ def bulkDM(request):
 
     return TemplateResponse(request, 'bulk_DM.html', context)
 
-def validate_number(user_id, twilio, phone, redis):
+
+def validate_number(user, twilio, phone, redis):
     validation = twilio.lookups.phone_numbers(phone).fetch(type=['caller-name', 'carrier'])
+    profile = user.profile
+    hash_number = hashlib.md5(phone.encode()).hexdigest()
```

Comment:
@danlipert didnt you want to use a diff hash function than md5?

Reply:
@apbendi Yup - using md5 here is too quick to iterate over the full set of USA numbers. We should use blowfish/bcrypt, probably with a salt as well: https://auth0.com/blog/hashing-in-action-understanding-bcrypt/

Reply:
The problem that I faced is if one user already registered a given number, in case we have 3000 salt numbers it will pretty expensive because the system needs to retrieve the 3000 numbers and encrypt and compare 3000 times. if that's ok, I can implement the logic for that. Also, any alternative solution is welcomed

Reply:
ah thats true - I suppose we could use the same salt for every entry but thats only marginally better. I'm fine with no salt

Reply:
You can use a deterministic salt based on username, in pseudo code:

```
salt = "gitcoin:" + username + ":"
value_to_hash = salt + phone_numer
```

so it might look like `gitcoin:apbendi:15558675209`

Reply:
Also since it looks like you're looking to increase the cost of a rainbow attack, you could do 100,000 rounds of bcrypt.

When I saw md5 I thought it was just being used for uniqueness checking.

Reply:
> is `phone` always guaranteed to be the numeric number? what if user 1 enters
> 
> > 123-456-7890
> 
> and the second user enterse
> 
> > (123)456-7890
> 
> wil they be diff numbers?

From the fronted is guaranteed to have the e.164 format independently if the user add parentheses, dashes or other phone related symbol https://www.twilio.com/docs/glossary/what-e164

Reply:
FYI for any onlookers. the latest version of this code uses this hash lib
```
    hash_number = hashlib.pbkdf2_hmac('sha256', phone.encode(), PHONE_SALT.encode(), 100000).hex()
```